.github/workflows/publish_charm.yaml

```diff
@@ -63,6 +63,53 @@ jobs:
           docker pull $image
           imageId=$(docker images $image --format "{{.ID}}")
           charmcraft upload-resource $charmName ${{ matrix.image }}-image --image=$imageId --verbosity=trace
+  lookup-external-images:
+    name: Lookup External Images
+    runs-on: ubuntu:latest
+    outputs:
+      images: ${{ env.IMAGES }}
+    steps:
+      - uses: actions/checkout@v3
+      - if: inputs.working-directory != './'
+        name: Change directory
+        run: |
+          TEMP_DIR=$(mktemp -d)
+          cp -rp ${{inputs.working-directory}}/. $TEMP_DIR
+          rm -rf .* * || :
+          cp -rp $TEMP_DIR/. .
+          rm -rf $TEMP_DIR
+      - name: Extract External Images
+        run: |
+          IMAGES=$(yq -c '.resources | to_entries | map(select(.value.type == "oci-image" and .value."upstream-source")) | map(.key)' metadata.yaml)
+          echo "External Images: $IMAGES"
+          echo "IMAGES=$IMAGES" >> $GITHUB_ENV
+  publish-external-images:
+    name: Push Externam Images to Charmhub
+    runs-on: ubuntu:latest
+    needs: [ lookup-external-images ]
+    if: ${{ needs.lookup-external-images.outputs.images != '[]' }}
+    strategy:
+      matrix:
+        image: ${{ fromJSON(needs.lookup-external-images.outputs.images) }}
+    steps:
+      - uses: actions/checkout@v3
+      - if: inputs.working-directory != './'
+        name: Change directory
+        run: |
+          TEMP_DIR=$(mktemp -d)
+          cp -rp ${{inputs.working-directory}}/. $TEMP_DIR
+          rm -rf .* * || :
+          cp -rp $TEMP_DIR/. .
+          rm -rf $TEMP_DIR
```

Comment:
I'm not sure to understand what this is doing, can you elaborate a bit ?

Reply:
This mostly for test purpose only. It was used when the required working directory is not the project root.

Reply:
Changing the working directory only affect following command in the same step. In the next step, everything will be reset.

Reply:
Yes I get it, but then why don't you change the folder in the same step as the upload-resource command ?